From a027a965dc0317d008cfff8e611516926f49a5a7 Mon Sep 17 00:00:00 2001
From: Wujian Sun <wujian.sun_1@nxp.com>
Date: Thu, 27 Jun 2024 14:43:08 +0800
Subject: [PATCH 06/13] LF-12406: Fixed webgl test fail for GL_MAX_SAMPLES
 checking

Differences Between WebGL and OpenGL ES 3.0 about GL_MAX_SAMPLES.

https://registry.khronos.org/OpenGL-Refpages/es3.0/html/glGetInternalformativ.xhtml

So disable the GL_MAX_SAMPLES checking for webgl context.

Upstream-Status: Inappropriate [i.MX specific]

Signed-off-by: Wujian Sun <wujian.sun_1@nxp.com>
---
 gpu/command_buffer/service/gles2_cmd_decoder.cc | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/gpu/command_buffer/service/gles2_cmd_decoder.cc b/gpu/command_buffer/service/gles2_cmd_decoder.cc
index 02078983a7b10..b581d3d05cdf8 100644
--- a/gpu/command_buffer/service/gles2_cmd_decoder.cc
+++ b/gpu/command_buffer/service/gles2_cmd_decoder.cc
@@ -7948,11 +7948,13 @@ bool GLES2DecoderImpl::ValidateRenderbufferStorageMultisample(
     }
   }
 
-  if (samples > renderbuffer_manager()->max_samples()) {
-    LOCAL_SET_GL_ERROR(
-        GL_INVALID_VALUE,
-        "glRenderbufferStorageMultisample", "samples too large");
-    return false;
+  if (!feature_info_->IsWebGLContext()) {
+    if (samples > renderbuffer_manager()->max_samples()) {
+      LOCAL_SET_GL_ERROR(
+          GL_INVALID_VALUE,
+          "glRenderbufferStorageMultisample", "samples too large");
+      return false;
+    }
   }
 
   if (width > renderbuffer_manager()->max_renderbuffer_size() ||
-- 
2.43.0

